homeassistant/components/remember_the_milk/entity.py

```diff
@@ -60,20 +60,21 @@ def create_task(self, call: ServiceCall) -> None:
             result = self._rtm_api.rtm.timelines.create()
             timeline = result.timeline.value
 
-            if hass_id is None or rtm_id is None:
+            if rtm_id is None:
                 result = self._rtm_api.rtm.tasks.add(
                     timeline=timeline, name=task_name, parse="1"
                 )
                 _LOGGER.debug(
                     "Created new task '%s' in account %s", task_name, self.name
                 )
-                self._rtm_config.set_rtm_id(
-                    self._name,
-                    hass_id,
-                    result.list.id,
-                    result.list.taskseries.id,
-                    result.list.taskseries.task.id,
-                )
+                if hass_id is not None:
```

Comment:
This check was added since it's a requirement to have `hass_id` be not `None` to get an rtm id from the config storage. Setting an rtm id with `hass_id` as `None` is useless since we would never retrieve the id in that case.